zwave_js_server/model/node/__init__.py

```diff
@@ -660,13 +660,15 @@ async def async_check_route_health(
         assert data
         return RouteHealthCheckSummary(data["summary"])
 
-    async def async_get_state(self) -> None:
+    async def async_get_state(self, update_internal_state: bool = True) -> NodeDataType:
         """Get node state."""
         data = await self.async_send_command(
             "get_state", require_schema=14, wait_for_result=True
         )
         assert data
-        self.update(data["state"])
+        if update_internal_state:
```

Comment:
Why shouldn't we always update the internal state? The retrieved state is correct, right?

Reply:
if we are getting the state from the driver for diagnostics, we don't want to update the internal state as it may cause the user to think that a problem they are trying to troubleshoot has been resolved when it hasn't been

Reply:
in most cases we will keep the default `update_internal_state == True`. We would only set it to false as part of device diagnostics

Reply:
I guess we could use the new get_node_unsafe command which effectively does the same thing without having to modify this command. Thoughts?

Reply:
Maybe we should remove the `update` call from this method and the library user can update the node state when needed instead after getting the node state?

Reply:
yeah we could do that. In HA this is only called in one place at the moment

Reply:
I think that is more explicit and composable.